akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -999,6 +999,25 @@ object Http extends ExtensionId[HttpExt] with ExtensionIdProvider {
     def whenTerminated: Future[HttpTerminated] =
       _whenTerminated.future
 
+    /**
+     * Adds this `ServerBinding` to the actor system's coordinated shutdown, so that [[unbind]] and [[terminate]] get
+     * called appropriately before the connection pools are shut down.
+     *
+     * @param hardTerminationDeadline timeout after which all requests and connections shall be forcefully terminated
+     */
+    def addToCoordinatedShutdown(hardTerminationDeadline: FiniteDuration)(implicit system: ClassicActorSystemProvider): ServerBinding = {
+      val shutdown = CoordinatedShutdown(system)
+      shutdown.addTask(CoordinatedShutdown.PhaseServiceUnbind, s"http-unbind-${localAddress}") { () =>
+        unbind()
+      }
+      shutdown.addTask(CoordinatedShutdown.PhaseServiceRequestsDone, s"http-terminate-${localAddress}") { () =>
+        terminate(hardTerminationDeadline).map(_ => Done)(ExecutionContexts.sameThreadExecutionContext)
+      }
+      shutdown.addTask(CoordinatedShutdown.PhaseServiceStop, s"http-shutdown-${localAddress}") { () =>
+        Http()(system).shutdownAllConnectionPools().map(_ => Done)(ExecutionContexts.sameThreadExecutionContext)
```

Comment:
Since this is about the client side, it seems weird that `addToCoordinatedShutdown` on the `ServerBinding` would cause those to be closed.

Shutting down the client connection pools seems less important anyway, perhaps we should just remove this task here and let the pools shut down along with the rest of the actor system? Or is there a scenario where that is messy?

Reply:
Good point.
I'll come up with separate information for the client. There, it is more up to the application to stop ongoing requests before they shut down the pools and the actor system.

Reply:
> Shutting down the client connection pools seems less important anyway, perhaps we should just remove this task here and let the pools shut down along with the rest of the actor system? Or is there a scenario where that is messy?

It's noisy because the actors are running and pulled out under them. But maybe the client side shutting down could be automatically added to coordinated shutdown. But that can be a different PR.